team/bundles/org.eclipse.jsch.ui/src/org/eclipse/jsch/internal/ui/preference/PreferencePage.java

```diff
@@ -305,6 +313,19 @@ private Control createKeyManagementPage(Composite parent){
 		gd.horizontalSpan=1;
 		keyLoad.setLayoutData(gd);
 
+		final Label keyGenerateSizeLabel=new Label(group, SWT.NONE);
+		keyGenerateSizeLabel.setText(Messages.CVSSSH2PreferencePage_148);
+		keyGenerateSize=new Spinner(group, SWT.BORDER);
+		keyGenerateSize.setValues(MIN_KEY_SIZE, MIN_KEY_SIZE, MAX_KEY_SIZE, MAX_KEY_SIZE_DIGIT_COUNT,
+				KEY_SIZE_INCREMENT, KEY_SIZE_INCREMENT);
+
+		keyGenerateSize.addKeyListener(new KeyAdapter() {
+			@Override
+			public void keyPressed(KeyEvent e) {
+				e.doit = false;
+			}
+		});
```

Comment:
This can be simplified using `KeyListener.keyPressedAdapter()`
```suggestion
		keySizeValue.addKeyListener(KeyListener.keyPressedAdapter(e -> e.doit = false));
```